### ISSUE TITLE
Feedback for OnePlus page

### ISSUE THREAD

**Issue by pascalpa** [urbandroid-team/dont-kill-my-app#180]
Hello. Your website is very interesting, but I disagree with some of your statements.
1) I own a OnePlus 7T and what you wrote about OnePlus should be updated. Maybe it was true in the past (it's my first OnePlus, so I don't know), but it is now untrue. Actually, OnePlus lets the user choose between 3 options, for every app: "Smart control" (default), "Optimize" or "Do not Optimize". And I can tell you that this system works perfectly fine. For example, my firewall (for which I selected "Do not optimize") constantly runs in the background, even after I reboot the phone. According to me, OnePlus does not deserve 5 thumbs down, but maybe 2 or 3 maximum.
2) On the opposite, I recently bought a Xiaomi Mi 9T Pro, that I sent back to the shop after a few days, for 3 reasons: a) it was constantly killing my firewall after every reboot and I did not find any solution to prevent this behaviour; b) it was too much bloated with crapware; c) Xiaomi's "Privacy policy" is a joke and does not respect the user's privacy. For me, this Xiaomi phone clearly deserves 5 thumbs down, or  even 6, if it was possible. ;-)
3) I agree on the fact that certain apps should be allowed to run in the background (if I want to get asleep with music, I need to have my player running). I also understand that a developer be very frustrated, when he (or she) sees that his/her app does not run the way it was intented to. 
4) However, as a user (and also as a sysadmin), I do need to have some control on my machines, no matter if it is a phone, a server, a PC or an iron. ;-) For example, I find very annoying, when I am listening to music, to hear it playing after having properly closed the app, and then, to be unable to stop it without poking around in the Settings. Sometimes, this is really getting on my nerves! Another example: as you know, some apps are very "talkative". They have the unpolite habit to send our data over the web, even when we are sleeping. I don't like this and I don't want this. For this reason, I sometimes use stuff like Hibernate or Kill App. I have even planned to use Greenify or Superfreeze, to avoid such annoyances. Fortunately, it seems that Oxygen OS does a good job in that matter, so I won't likely be forced to install 3rd party apps. 
5) To summarize things: when I close an app, I would like to have the choice between completely closing it, or letting it run in the background. I would also like to be able to decide whenever an app should automatically start with the system, or not. (I would also like to really have the choice between a "googled" Android and a free OS... but it's another story.)
Thank you for your attention.
Pascal (from France)

**Comment by petrnalevka** [comment]
Hello Pascal, many thanks for the feedback. Here you can find how we do the scoring of individual  OEM.. https://dontkillmyapp.com/about_score

The reason why 1+ is getting 5/5 is that we are getting many reports that even when you make apps not optimized after a system update 1+ reverts your settings. So this in our opinion prevents users from reliably setting the system background processing behavior. 

Moreover we are maintaining apps with millions of installs which do background processing on various devices. And this results in hundreds of background processing support requests daily. This gives us an opportunity to map how is each OEM doing. 

The fact is most or second most common complains are from users of  1+ devices even their market share is tiny in comparison to others OEMs. This makes the cost of supporting a 1+ user bigger than any for any other OEM.

We cannot say for sure if this has improved somehow with latest 1+ models or not - definitely not with 1+6 devices - I'm quite sure there. But it is evident that users do not understand how to configure their background processes on 1+ phone an this is why 1+ deserves this score IMHO.

**Comment by thubalek** [comment]
I second that. In last months One Plus users were most requesting support (it means most generating support costs) because my long term running apps were killed by their optimization. I even proposed to make their rating even worse (https://github.com/urbandroid-team/dont-kill-my-app/issues/135) :smile:

**Comment by IRainman** [comment]
@petrnalevka Hello, thank U.

I already have 1+3T, and recently I replaced it to 1+7Pro and... it's very bad device. 
1) I set no optimisation for app
2) I completely disable smart memory
3) I pin all app why I need in active app
and nothing helped, everything is done, but still does not work. Stupid Oxygen OS killed and killed apps, widgets not working on desktop(!), and other bad behavior happens (notifications sometimes come with a delay in hours if the screen is off).

I never buy 1+ again, only Google now.

I set -10 stars to 1+, now it's completely shit :(

**Comment by petrnalevka** [comment]
Yes, personally after the horrors we have with supporting 1+ devices I really could not recommend to buy a 1+ to anyone..even my worst enemy :)

**Comment by IRainman** [comment]
P.S. on the 1+7Pro only one app and only in foreground may work normally and only when screen is on. No other settings help at all, any application running not in this scenario can be killed by OS.

taaddam :) I think I now have a good mobile router for a monstrous cost, since using this horror is otherwise impossible.

**Comment by IRainman** [comment]
> 
> 
> I think it would also be a good idea to note which OOS versions issues apply to, for example "Deep optimization" isn't even a thing on modern OOS anymore.

I use 1+7Pro, the widgets for smart lamp working only after two tap! First tap is needed for starting app, second for app reacting to the tap, this is shit. Oxygen OS from 1+ is completely crap.

**Comment by Xalaxis** [comment]
> I use 1+7Pro, the widgets for smart lamp working only after two tap! First tap is needed for starting app, second for app reacting to the tap, this is shit. Oxygen OS from 1+ is completely crap.

That may just be a problem with your particular app, and notably is unlikely to be related to background battery optimizations (tapping the widget should start the app on it's own).  I use the Hue app, and have no issues with light widgets on my OnePlus 6.

**Comment by Xalaxis** [comment]
>    (notifications sometimes come with a delay in hours if the screen is off).

Are you sure that's not just Doze behavior?

**Comment by IRainman** [comment]
> Are you sure that's not just Doze behavior?
I don't sure. But this behavior not change when device on the power charger.
